### PR TITLE
[terra] Fix terra on azure helm chart and make targets

### PR DESCRIPTION
**Pull request by daniel-goldstein**
For Hail Batch on Terra Azure, the production artifact is Helm chart containing the necessary kubernetes resources to run a Hail Batch deployment in a Terra k8s cluster. This deployment contains slightly modified containers of the batch front-end, batch driver and a mysql database. This chart is currently built manually using the targets in `batch/terra-chart/Makefile`. As this process is not currently automatically tested, it's very prone to bit rot. This PR is an amalgamation of fixes that I needed to make to get `main` to build in the current Terra. A non-exhaustive list of the changes are:

- After changing from gradle to mill, the some Dockerfiles and make targets needed to change to account for the new location of the JAR.
- I removed some redundancy in invocations of `docker build` by relying on the generic targets that we now have in the top level Makefile.
- Terra changed how they handle identity management for the kubernetes deployment, from `aadpodidentity` to `workloadIdentity`. This changes the chart to work with their new inputs they provide.

Ultimately, terra should have a CI system that we can push charts to and receive feedback on whether it passed our test suite in their test environment.

### CODE REVIEW

**Comment by daniel-goldstein** [comment]
@iris-garden Let me know if you want to meet to get a low-down on what's happening here, I realize I'm asking for a review on something only dan and I ever really laid eyes on.

File: Makefile [hunk 1]
```diff
@@ -220,6 +220,17 @@ hailgenetics-vep-grch38-95-image: hail-ubuntu-image
 		--build-arg BASE_IMAGE=$(shell cat hail-ubuntu-image)
 	echo $(IMAGE_NAME) > $@
 
+terra-batch-image: batch-image
+	./docker-build.sh . batch/terra-chart/Dockerfile.batch $(IMAGE_NAME) \
+		--build-arg BASE_IMAGE=$(shell cat batch-image)
+	echo $(IMAGE_NAME) > $@
+
+terra-batch-worker-image: batch-worker-image
+	$(MAKE) -C hail shadowJar
```
Comment (ehigham) [inline Makefile:229]: Are you using `HAIL_RELEASE_MODE=1` somewhere? Might cause this to use a very slow version of hail in terra.

Comment (daniel-goldstein) [reply Makefile:229]: 🤦 thank you for reminding me that this flipped

**Review by iris-garden — DISMISSED** [review]
hey! so sorry for the late review. i don't see anything particularly confusing in the PR content, just have a quick question: when you say you made these changes to get `main` building on the current Terra, does that mean you've already tested that that build process works with these changes? just wanted to double check, but if so, happy to approve!

**Review by iris-garden — APPROVED** [review]
sounds good to me!

**Pull request merged**